pages/docs/concepts/asyncapi-document/adding_messages.md

```diff
@@ -0,0 +1,137 @@
+---
+title: Adding Messages
+weight: 140
+---
+
+[Messages](../message.md) in AsyncAPI represent the data being transmitted between your systems. They always include a payload, which is the data the application is sending or receiving, and they may also include headers, which are used to provide additional context or information about the message exchange.
+
+```mermaid
+graph TD;
+    A[AsyncAPI Message] --> B[Headers]
+    A --> C[Payload]
+    B --> D[Header Content]
+    C --> E[Payload Content]
+
+    style A fill:#47BCEE,stroke:#47BCEE;
+```
+
+Here is an example of a simple message:
+
+```yml
+channels:
+    user/signedup:
+    subscribe:
+        message:
+        $ref: "#/components/messages/userSignUp"
+messages:
+    userSignUp:
+    summary: Action to sign a user up.
+    description: |
+        Multiline description of what this action does.
+        Here you have another line.
+```
+
+This document defines a `user/signedup` channel where a `subscription` to the `userSignUp` message can be made; with this `subscription`.
+
+## Using `oneOf` for Multiple Message Types
+
+AsyncAPI supports the use of `oneOf` for messages under operations, allowing you to define multiple possible message types for a single operation.
+
+```mermaid
+graph TD;
+    A[Operations] --> B[Messages]
+    B --> C[oneOf]
+    C --> D[Message Type 1]
+    C --> E[Message Type 2]
+
+    style B fill:#47BCEE,stroke:#47BCEE;
+    style D fill:#47BCEE,stroke:#47BCEE;
+    style E fill:#47BCEE,stroke:#47BCEE;
+```
+
+Here is an example document of how `oneOf` permits the use of multiple message types for a single operation:
+
+```yml
+channel:
+$ref: '#/channels/userSignupReply'
+messages:
+oneOf:
+    - $ref: '#/components/messages/userSignedUp'
+    - $ref: '#/components/messages/userSignedUpReply'
+```
+
+The code denotes a channel `userSignupReply` under which two messages can be sent or received: `userSignedUp` or `userSignedUpReply`.
+
+## Specifying `contentType` in Messages
+
+The `contentType` field specifies the format of the payload. If it's not provided, the default payload format is `application/json`. However, it's recommended always to specify the contentType for clarity and to avoid potential misinterpretations.
```

Comment:
```suggestion
The `contentType` field specifies the format of the payload. If it's not provided, the default payload format is `application/json`. However, it's recommended to always specify the contentType for clarity and to avoid potential misinterpretations.
```
How about this?